llvm/lib/Target/AMDGPU/AMDGPUSwLowerLDS.cpp

```diff
@@ -655,20 +654,37 @@ void AMDGPUSwLowerLDS::getLDSMemoryInstructions(
       } else if (AtomicCmpXchgInst *XCHG = dyn_cast<AtomicCmpXchgInst>(&Inst)) {
         if (XCHG->getPointerAddressSpace() == AMDGPUAS::LOCAL_ADDRESS)
           LDSInstructions.insert(&Inst);
+      } else if (AddrSpaceCastInst *ASC = dyn_cast<AddrSpaceCastInst>(&Inst)) {
+        if (ASC->getSrcAddressSpace() == AMDGPUAS::LOCAL_ADDRESS &&
+            ASC->getDestAddressSpace() == AMDGPUAS::FLAT_ADDRESS)
+          LDSInstructions.insert(&Inst);
       } else
         continue;
     }
   }
 }
 
-Value *
-AMDGPUSwLowerLDS::getTranslatedGlobalMemoryGEPOfLDSPointer(Value *LoadMallocPtr,
+Value *AMDGPUSwLowerLDS::getTranslatedGlobalMemoryPtrOfLDS(Value *LoadMallocPtr,
                                                            Value *LDSPtr) {
   assert(LDSPtr && "Invalid LDS pointer operand");
+  Type *LDSPtrType = LDSPtr->getType();
+
+  if (LDSPtrType->isVectorTy()) {
+    // Handle vector of pointers
+    VectorType *VecPtrTy = cast<VectorType>(LDSPtrType);
+    ElementCount NumElements = VecPtrTy->getElementCount();
+    Type *Int32VecTy = VectorType::get(IRB.getInt32Ty(), NumElements);
```

Comment:
You can use DL.getIntPtrType for this instead of hardcoding the address space size